node/node.go

```diff
@@ -200,6 +200,15 @@ func (n *Node) headerPublishLoop(ctx context.Context) {
 			if err != nil {
 				n.Logger.Error("failed to gossip block header", "error", err)
 			}
+		case commit := <-n.blockManager.CommitOutCh:
+			commitBytes, err := commit.MarshalBinary()
+			if err != nil {
+				n.Logger.Error("failed to serialize block commit", "error", err)
+			}
+			err = n.P2P.GossipCommit(ctx, commitBytes)
+			if err != nil {
+				n.Logger.Error("failed to gossip block commit", "error", err)
+			}
```

Comment:
Btw, it's not necessary to make sync loop only responsible for publishing/gossiping messages, and the code that needs to publish can use GossipCommit directly. But I think you understand this and made `CommitOutCh` for symmetry with `CommitInCh` 😁  which is fine. However, it's not the best justification for bottlenecking both msg publishing and the syncloop.

Reply:
Good catch! I missed this bit during refactoring :facepalm: